### ISSUE TITLE
Emphasize that servers support exposure of doc strings over LabRAD

### ISSUE THREAD

**Issue by DanielSank** [labrad/pylabrad#196]
Where should we put this information? Where do we document the fact that LabRAD supports server setting documentation?

@maffoo thoughts?

**Comment by ejeffrey** [comment]
More to the point, none of the scala servers have documentation strings.
This requirement needs to be added to the style guide.  Pylabrad
automatically uses the method docstring, which should be there anyway (and
is already in the style guide).

**Comment by DanielSank** [comment]
> pylabrad automatically uses the method docstring, which should be there anyway (and
is already in the style guide).

Good point. I've just discovered that in scalabrad there's a `doc` argument to the `@setting` decorator. The string passed into `doc` shows up as the LabRAD doc string. I can't tell yet if there's a way to access documentation associated to a scala function (cursory Google searches indicate not).

> More to the point, none of the scala servers have documentation strings. This requirement needs to be added to the style guide.

+1

**Comment by maffoo** [comment]
The pylabrad setting decorator is documented fairly thoroughly in the wiki: https://github.com/labrad/pylabrad/wiki/Writing-Servers#setting-metadata. We could certainly add more detail or emphasize certain things there.

It's hard to know what's most important to emphasize, of course. The last you happened not to know and just figured out after a lot of digging usually tends to seem like the most important thing to emphasize. But the next thing will seem just as important :)

**Comment by DanielSank** [comment]
I was thinking more in the LabRAD repo.

**Comment by ejeffrey** [comment]
I don't think the documentation is particularly lacking.  I just think we
need the example servers to actually do it.  I can't speak for everyone,
but if I want to write a server, the first thing I am going to look for is
an example to follow, rather than proactively search documentation for a
feature I don't know about.  Documentation should be there for those things
that you say "I wonder how you do X"

**Comment by DanielSank** [comment]
An example server that does this counts as documentation to me :)

**Comment by maffoo** [comment]
The [test_server](https://github.com/labrad/pylabrad/blob/master/labrad/servers/test_server.py) has docstrings on _almost_ all of its remotely-callable settings. Is there anything in particular that you think is missing there?

**Comment by DanielSank** [comment]
Actually, now that I think about it, the best place for this information is probably in the setting decorator and LabradServer class.

**Comment by maffoo** [comment]
You mean a note in the LabradServer class docstring indicating that the docstring given to subclasses will be visible to labrad clients?

**Comment by DanielSank** [comment]
Yeah, and similarly that the docstrings given to functions wrapped by `@setting` are exposed as well. This should *probably* be noted in the wiki entry on writing servers too, because I bet almost nobody reads the code for `LabradServer` and/or `setting`.
